File: app/src/androidTest/java/com/przemolab/oknotifier/modules/OpenKattisServiceTests.java
package com.przemolab.oknotifier.modules;

import android.support.test.runner.AndroidJUnit4;

import com.przemolab.oknotifier.models.Contest;
import com.przemolab.oknotifier.models.Contestant;

import org.junit.Test;
import org.junit.runner.RunWith;

import java.util.List;

import static junit.framework.Assert.assertNotNull;

@RunWith(AndroidJUnit4.class)
public class OpenKattisServiceTests {

    @Test
    public void getOngoingContests_returnsContestsList() {
        // given
        OpenKattisService service = new OpenKattisService();

        // when
        List<Contest> result = service.getOngoingContests();

        // then
        assertNotNull(result);
    }

    @Test
    public void getContestStandings_returnsContestantsList() {
        // given
        OpenKattisService service = new OpenKattisService();
        List<Contest> contests = service.getOngoingContests();

        if (!contests.isEmpty()) {
            // when
            List<Contestant> result = service.getContestStandings(contests.get(0).getId());

            // then
            assertNotNull(result);
        }
    }
}

File: app/src/androidTest/java/com/przemolab/oknotifier/modules/OpenKattisServiceTests.kt
package com.przemolab.oknotifier.modules

import android.support.test.runner.AndroidJUnit4

import org.junit.Test
import org.junit.runner.RunWith

import junit.framework.Assert.assertNotNull

@RunWith(AndroidJUnit4::class)
class OpenKattisServiceTests {

    @Test
    fun getOngoingContests_returnsContestsList() {
        // given
        val service = OpenKattisService()

        // when
        val result = service.ongoingContests

        // then
        assertNotNull(result)
    }

    @Test
    fun getContestStandings_returnsContestantsList() {
        // given
        val service = OpenKattisService()
        val contests = service.ongoingContests

        if (!contests!!.isEmpty()) {
            // when
            val result = service.getContestStandings(contests[0].contestId)

            // then
            assertNotNull(result)
        }
    }
}
